### PR TITLE
Chess prototype

### PR DESCRIPTION
**Pull request by namin**
Hi,

Thanks for this great work. I love how everything is pluggable.

I don't claim this is ready as is for merging, but just wanted to let you know that I am interested in contributing a small chess implementation.

API changes:
- I added an optional `toArray` in the `Game` interface so that the board representation can be more general for self play and just converted to an array for the neural network.
- I noticed that Arena calls `getGameEnded` passing player 1 instead of the current player. This seems OK, but just wanted to amend the documentation.

Chess:
- I called the directory _chess instead of chess because I am using python-chess via `import chess`...
- I am not sure at all about the neural network. Just changed from conv2d to conv3d in the most obvious way.
- I am still testing whether I can get any interesting results.

(The changes to main.py are FYI, not to be merged.)

Thanks again for this work. It's been so much fun and enlightening to play with it.

### CODE REVIEW

**Comment by namin** [comment]
Note that this doesn't work at all, because it never seems to get out of draw in self play phase.

**Comment by namin** [comment]
I will close this pull request as I refine the code further first, but curious to hear if you have any thoughts already.

Thanks and sorry for the noise.

**Comment by suragnair** [comment]
Hi Nada

Thanks so much for taking the time to take a deep dive into the code and write up a chess implementation. It would definitely be a super useful addition to the repository.

The API change seems fine, the only caveat is that files for other games will need to be modified to add that method (should be just 2-3 lines). 

Would be worth testing it to see if it gets out of the draws in self-play. If it doesn't, it might be due to a bug either in the game logic or in the neural network architecture or training. 

Good luck and I'm looking forward to the chess implementation!

File: Coach.py [hunk 1]
```diff
@@ -123,7 +123,7 @@ def learn(self):
             pwins, nwins, draws = arena.playGames(self.args.arenaCompare)
 
             print('NEW/PREV WINS : %d / %d ; DRAWS : %d' % (nwins, pwins, draws))
-            if pwins+nwins == 0 or float(nwins)/(pwins+nwins) < self.args.updateThreshold:
+            if (not (pwins+nwins == 0)) and (float(nwins)/(pwins+nwins) < self.args.updateThreshold):
```
Comment (Unimax) [inline Coach.py]: why? this change?

File: Coach.py [hunk 2]
```diff
@@ -51,7 +51,7 @@ def executeEpisode(self):
             pi = self.mcts.getActionProb(canonicalBoard, temp=temp)
             sym = self.game.getSymmetries(canonicalBoard, pi)
             for b,p in sym:
-                trainExamples.append([b, self.curPlayer, p, None])
+                trainExamples.append([self.game.toArray(b), self.curPlayer, p, None])
```
Comment (Unimax) [inline Coach.py]: does this mean all games have this function now?

File: Coach.py (hunk 1 above)
Comment (suragnair) [reply Coach.py]: This PR has not been merged.

**Pull request closed without merging**